### PR TITLE
Better logging to the LHM gem

### PR DESCRIPTION
**Pull request by Manan007224**
LHM gem earlier did not print much of the log messages via the logger earlier. All it printed was the start of the class (Chunker, Entangler, AtomicSwitcher etc). 

In this PR I am adding the code so that LHM logs everything via the `logger` and not with `puts`. 

An example output of the LHM run of table to add a column is 
```
I, [2021-09-15T08:59:55.502850 #50456]  INFO -- : Starting LHM run on table=lhmn_table1
I, [2021-09-15T08:59:55.507802 #50456]  INFO -- : Starting run of class=Lhm::Migrator
I, [2021-09-15T08:59:55.516547 #50456]  INFO -- : Created destination table lhmn_table1
I, [2021-09-15T08:59:55.524681 #50456]  INFO -- : Starting run of class=Lhm::Entangler
I, [2021-09-15T08:59:55.534139 #50456]  INFO -- : Created triggers on #<Lhm::Table:0x00007fe6180af630>
I, [2021-09-15T08:59:55.535758 #50456]  INFO -- : Starting run of class=Lhm::Chunker
I, [2021-09-15T08:59:55.557892 #50456]  INFO -- : Inserted 2000 rows into the destination table from 1 to 2000
I, [2021-09-15T08:59:55.557944 #50456]  INFO -- : Starting run of class=Lhm::Throttler::Time
I, [2021-09-15T08:59:55.662146 #50456]  INFO -- : 57.17% (2001/3500) complete
I, [2021-09-15T08:59:55.683339 #50456]  INFO -- : Inserted 1500 rows into the destination table from 2001 to 3500
I, [2021-09-15T08:59:55.683390 #50456]  INFO -- : Starting run of class=Lhm::Throttler::Time
I, [2021-09-15T08:59:55.786139 #50456]  INFO -- : 100% complete
I, [2021-09-15T08:59:55.787847 #50456]  INFO -- : Starting run of class=Lhm::AtomicSwitcher
I, [2021-09-15T08:59:55.807536 #50456]  INFO -- : Dropped triggers on #<Lhm::Table:0x00007fe6180af630>
```

Some extra changes - I have modified the `Printer` object so that it logs all the messages via the logger.

### CODE REVIEW

File: lib/lhm/chunker.rb [hunk 1]
```diff
@@ -47,15 +47,19 @@ def execute
         affected_rows = ChunkInsert.new(@migration, @connection, bottom, top, @options).insert_and_return_count_of_rows_created
         expected_rows = top - bottom + 1
 
+        Lhm.logger.info("Inserted #{affected_rows} rows into the destination table from #{bottom} to #{top}")
```
Comment (shuhaowu) [inline lib/lhm/chunker.rb:50]: I wouldn't log every iteration. Maybe at least once and then once every X minutes.

**Review by shuhaowu — COMMENTED** [review]
Looking good. Feedback on some minor comments and typos are attached. Once addressed I think we're good to merge.

File: lib/lhm/printer.rb [hunk 2]
```diff
@@ -42,7 +40,7 @@ def write(message)
           extra = 0
         end
 
-        @output.write "\r#{message}" + (' ' * extra)
```
[the remark this answers is not in the document]
Comment (Manan007224) [reply lib/lhm/printer.rb:45]: I am okay to remove this class, but there is a class called `Dot` which extends from `Output`. Here is the definition of this class - https://github.com/Shopify/lhm/blob/master/lib/lhm/printer.rb#L49. 

The `Dot` class does not seem to be used anywhere in the LHM codebase, so we can too remove that class as well.

File: lib/lhm/cleanup/current.rb [hunk 3]
```diff
@@ -63,6 +63,8 @@ def execute_ddls
             retriable_connection.execute(ddl)
           end
         end
+        Lhm.logger.info("Dropped triggers on #{@lhm_triggers_for_origin.join(', ')}")
+        Lhm.logger.info("Dropeed tables #{@lhm_triggers_for_origin.join(', ')}")
```
[the remark this answers is not in the document]
Comment (Manan007224) [reply lib/lhm/cleanup/current.rb:67]: I would prefer these messages in two separate lines as I am also printing the names of the triggers and tables dropped.

File: lib/lhm/printer.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (shuhaowu) [reply lib/lhm/printer.rb:45]: Dot might be used during local execution?

File: lib/lhm/cleanup/current.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (shuhaowu) [reply lib/lhm/cleanup/current.rb]: That's fine. There's a small typo to fix then.

File: lib/lhm/chunker.rb [hunk 4]
```diff
@@ -100,5 +106,14 @@ def validate
       return if @chunk_finder.table_empty?
       @chunk_finder.validate
     end
+
+    # Only log the chunker progress every 5 minutes instead of every iteration
+    def log_progress
```
Comment (shuhaowu) [inline lib/lhm/chunker.rb:111]: Maybe rate_limited_log_progress

File: lib/lhm/printer.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (Manan007224) [reply lib/lhm/printer.rb:45]: In that case, let's leave these classes as it is.

[the remark this answers is not in the document]
Comment (shuhaowu) [reply lib/lhm/printer.rb:45]: Lets do this in a separate PR.

**Review by shuhaowu — APPROVED** [review]

**Pull request merged**